### PR TITLE
fix(appsec): run the AppSec processor before the writer

### PR DESCRIPTION
**Pull request by nizox**
This PR inserts the AppSec processor before the `SpanAggregator` processor. In fact, the writer runs at the end of this aggregator so any changes to the spans done after this processor won't be sent to the agent.

### CODE REVIEW

**Review by vdeturckheim — APPROVED** [review]

**Comment by brettlangdon** [comment]
should we have a test asserting the order of the processors, since the ordering matters?

**Comment by nizox** [comment]
> should we have a test asserting the order of the processors, since the ordering matters?

A unit test could asserts that the writer receives the span after the AppSec processor, but that would be very specific. The system tests should already fail on this issue.

File: ddtrace/tracer.py [hunk 1]
```diff
@@ -681,7 +681,7 @@ def _initialize_span_processors(self, appsec_enabled=asbool(get_env("appsec", "e
                 from .appsec.processor import AppSecSpanProcessor
 
                 appsec_span_processor = AppSecSpanProcessor()
-                self._span_processors.append(appsec_span_processor)
+                self._span_processors.insert(0, appsec_span_processor)
```
Comment (Kyle-Verhoog) [inline ddtrace/tracer.py:684]: I'd rather we initialize processors in order (something like how we do the trace processors)

This way it's more clear that `SpanAggregator` is the last in the list.

**Review by Kyle-Verhoog — COMMENTED** [review]
Could we add a snapshot test case to ensure the appsec tags are set properly?

File: ddtrace/tracer.py (hunk 1 above)
Comment (brettlangdon) [reply ddtrace/tracer.py:684]: for example:

```python
processors = []

if app_sec_enabled:
    processors.append(AppSecSpanProcessor())

processors.append(SpanAggregator())
```

**Comment by vdeturckheim** [comment]
closing in favor of #3239 as this one is from a fork

**Pull request closed without merging**